probatus/feature_elimination/feature_elimination.py

```diff
@@ -955,16 +935,244 @@ def __init__(
+                sample weighting (check the corresponding scikit-learn documentation).
+                Array of weights that are assigned to individual samples.
+                Note that they're only used for fitting of  the model, not during evaluation of metrics.
+                If not provided, then each sample is given unit weight.
+
+            train_index (np.array):
+                Positions of train folds samples.
+
+            val_index (np.array):
+                Positions of validation fold samples.
+
+        Raises:
+            ValueError: if the clf is not supported.
+
+        Returns:
+            dict: fit parameters
+        """
+
+        cat_features = [col for col in X_train.select_dtypes(include=["category"]).columns]
+        fit_params = {
+            "X": Pool(X_train, y_train, cat_features=cat_features),
+            "eval_set": Pool(X_val, y_val, cat_features=cat_features),
+            "early_stopping_rounds": self.early_stopping_rounds,
+            # Evaluation metric should be passed during initialization
+        }
+        if sample_weight is not None:
+            fit_params["X"].set_weight(sample_weight.iloc[train_index])
+            fit_params["eval_set"].set_weight(sample_weight.iloc[val_index])
+        return fit_params
+
+    def _get_fit_params(
+        self, clf, X_train, y_train, X_val, y_val, sample_weight=None, train_index=None, val_index=None
+    ):
+        """Get the fit parameters for the specified classifier.
+
+        Args:
+            clf (binary classifier):
+                Model to be fitted on the train folds.
+
+            X_train (pd.DataFrame):
+                Train Dataset used in CV.
+
+            y_train (pd.Series):
+                Train Binary labels for X.
+
+            X_val (pd.DataFrame):
+                Validation Dataset used in CV.
+
+            y_val (pd.Series):
+                Validation Binary labels for X.
+
+            sample_weight (pd.Series, np.ndarray, list, optional):
+                array-like of shape (n_samples,) - only use if the model you're using supports
+                sample weighting (check the corresponding scikit-learn documentation).
+                Array of weights that are assigned to individual samples.
+                Note that they're only used for fitting of  the model, not during evaluation of metrics.
+                If not provided, then each sample is given unit weight.
+
+            train_index (np.array):
+                Positions of train folds samples.
+
+            val_index (np.array):
+                Positions of validation fold samples.
+
+        Raises:
+            ValueError: if the clf is not supported.
+
+        Returns:
+            dict: fit parameters
+        """
+        # The lightgbm and xgboost imports are temporarily placed here, until the tests on
+        # macOS have been fixed.
+        from lightgbm import LGBMModel
+        from xgboost.sklearn import XGBModel
+
```

Comment:
Can you get the import here for catboost? instead of the top of the file

Reply:
The import has been moved here